src/grpc_server.cc

```diff
@@ -3681,9 +3681,12 @@ ModelInferHandler::Process(InferHandler::State* state, bool rpc_ok)
     }
     // Get request ID for logging in case of error.
     const char* request_id = nullptr;
-    LOG_TRITONSERVER_ERROR(
-        TRITONSERVER_InferenceRequestId(irequest, &request_id),
-        "unable to retrieve request ID string");
+    if (err == nullptr) {
```

Comment:
This is opposite to the comment at line 3682. Do you mean `if (irequest != nullptr)`?

Reply:
Good catch! Fixed.